examples/sphDiagramGeneration/sph-diagram-generation.cpp

```diff
@@ -66,57 +66,115 @@ int main(int argc, char *argv[]) {
       boxMin, boxMax, cutoff);
 
+  autopas::sph::SPHCalcDensityFunctor densfunc;
+
+  autopas::sph::SPHCalcHydroForceFunctor hydrofunc;
+
   int numParticles = 16;
   int numIterations = 100000;
   int whichContainer = 0;
-  // int whichFunctor = 0;
-  if (argc == 5 or argc == 4) {
+  int whichFunctor = 0;
+  double skin = 0.;
+  int rebuildFrequency = 10;
+  if (argc == 7) {
+    numParticles = atoi(argv[1]);
+    numIterations = atoi(argv[2]);
+    whichContainer = atoi(argv[3]);
+    whichFunctor = atoi(argv[4]);
+    skin = atof(argv[5]);
+    rebuildFrequency = atof(argv[6]);
+  } else if (argc == 5) {
+    numParticles = atoi(argv[1]);
+    numIterations = atoi(argv[2]);
+    whichContainer = atoi(argv[3]);
+    whichFunctor = atoi(argv[4]);
+  } else if (argc == 4) {
     numParticles = atoi(argv[1]);
     numIterations = atoi(argv[2]);
     whichContainer = atoi(argv[3]);
-    //  whichFunctor = atoi(argv[4]);
   } else {
     exit(1);
   }
 
+  autopas::VerletLists<autopas::sph::SPHParticle> verletCont(boxMin, boxMax, cutoff, skin * cutoff, rebuildFrequency);
+
   addParticles(lcCont, numParticles);
 
   for (auto it = lcCont.begin(); it.isValid(); ++it) {
     dirCont.addParticle(*it);
+    verletCont.addParticle(*it);
   }
 
   if (whichContainer == 0) {
-    measureContainer(&lcCont, numParticles, numIterations);
+    if (whichFunctor == 0) {
+      measureContainer(&lcCont, &densfunc, numParticles, numIterations);
+    } else if (whichFunctor == 1) {
+      measureContainer(&lcCont, &hydrofunc, numParticles, numIterations);
+    } else {
+      std::cout << "wrong functor given" << std::endl;
+      exit(2);
+    }
   } else if (whichContainer == 1) {
-    measureContainer(&dirCont, numParticles, numIterations);
+    if (whichFunctor == 0) {
+      measureContainer(&dirCont, &densfunc, numParticles, numIterations);
+    } else if (whichFunctor == 1) {
+      measureContainer(&dirCont, &hydrofunc, numParticles, numIterations);
+    } else {
+      std::cout << "wrong functor given" << std::endl;
+      exit(2);
+    }
+  } else if (whichContainer == 2) {
+    if (whichFunctor == 0) {
+      measureContainer(&verletCont, &densfunc, numParticles, numIterations);
+    } else if (whichFunctor == 1) {
+      measureContainer(&verletCont, &hydrofunc, numParticles, numIterations);
+    } else {
+      std::cout << "wrong functor given" << std::endl;
+      exit(2);
+    }
+  } else {
+    std::cout << "invalid container id" << std::endl;
+    exit(3);
   }
 }
 
-template <class Container>
-void measureContainer(Container *cont, int numParticles, int numIterations) {
-  autopas::sph::SPHCalcDensityFunctor func;
-  autopas::FlopCounterFunctor<autopas::sph::SPHParticle, autopas::FullParticleCell<autopas::sph::SPHParticle>>
-      flopFunctor(cont->getCutoff());
+template <class Container, class Functor>
+void measureContainer(Container *cont, Functor *func, int numParticles, int numIterations) {
```

Comment:
add `TODO: fix flopFunctor for verlet` or we might delete this by accident
(at least this was the problem if I remember corretly?)

Reply:
hmm I'm not a huge fan of the flopcounters :p 
they are only working for fixed cutoff radii and don't really show what is happening.
MMUPS, etc. are way more meaningful and they sort of force you to compare against other codes...

the flops are only a theoretical value and hold no real comparability against other codes (each code counts flops differently). e.g.
* ls1: only flops that actually need to be computed are counted (kernel calls only counted if `distance < cutoff` ) -> less flops are counted than actually calculated, i.e. you don't count unnecessary flop calculations that didn't need to be calculated
* other codes: count all the flops actually performed: even if they result from vectorization and masking, i.e. independent of whether the distance is smaller than the cutoff. -> makes everything appear way (~8x) faster.